test/sslapitest.c

```diff
@@ -1936,17 +2046,27 @@ static int test_tlsext_status_type(void)
      * Now actually do a handshake and check OCSP information is exchanged and
      * the callbacks get called
      */
-    SSL_CTX_set_tlsext_status_cb(cctx, ocsp_client_cb);
-    SSL_CTX_set_tlsext_status_arg(cctx, &cdummyarg);
-    SSL_CTX_set_tlsext_status_cb(sctx, ocsp_server_cb);
-    SSL_CTX_set_tlsext_status_arg(sctx, &cdummyarg);
+    if (multi_stapling) {
+        cdummyarg = 1;
+        SSL_CTX_set_tlsext_status_cb(cctx, ocsp_client_cb_multi);
+        SSL_CTX_set_tlsext_status_arg(cctx, &cdummyarg);
+        SSL_CTX_set_tlsext_status_cb(sctx, ocsp_server_cb_multi);
+        SSL_CTX_set_tlsext_status_arg(sctx, &cdummyarg);
+    } else {
+        cdummyarg = 1;
+        SSL_CTX_set_tlsext_status_cb(cctx, ocsp_client_cb_single);
+        SSL_CTX_set_tlsext_status_arg(cctx, &cdummyarg);
+        SSL_CTX_set_tlsext_status_cb(sctx, ocsp_server_cb_single);
+        SSL_CTX_set_tlsext_status_arg(sctx, &cdummyarg);
+    }
     if (!TEST_true(create_ssl_objects(sctx, cctx, &serverssl,
                                       &clientssl, NULL, NULL))
             || !TEST_true(create_ssl_connection(serverssl, clientssl,
                                                 SSL_ERROR_NONE))
             || !TEST_true(ocsp_client_called)
             || !TEST_true(ocsp_server_called))
         goto end;
+    SSL_set_tlsext_status_ocsp_resp(serverssl, NULL, 0);
```

Comment:
`SSL_free()` should free the OCSP response, no ?

Reply:
The OCSP responses are freed in ossl_ssl_connection_free